File: app/src/main/java/app/galentin/ru/nonograms/MainActivity.java
package app.galentin.ru.nonograms;

import android.content.Intent;
import android.graphics.Typeface;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button startButton = (Button)findViewById(R.id.playButton);
        Button rulesButton = (Button)findViewById(R.id.rulesButton);
        View.OnClickListener onClickListener = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                switch (v.getId()){
                    case R.id.playButton :
                        try{
                            Intent intent = new Intent(MainActivity.this, SizeNonograms.class);
                            startActivity(intent);
                        }catch (Exception e) {
                            e.printStackTrace();
                        }
                        break;
                    case R.id.rulesButton :
                        try{
                            Intent intent = new Intent(MainActivity.this, Rules.class);
                            startActivity(intent);
                        }catch (Exception e) {
                            e.printStackTrace();
                        }
                        break;
                }
            }
        };
        startButton.setOnClickListener(onClickListener);
        rulesButton.setOnClickListener(onClickListener);

        TextView nameTextView=(TextView)findViewById(R.id.gameName);
        Typeface typeFace=Typeface.createFromAsset(getAssets(),"EastAP.ttf");
        nameTextView.setTypeface(typeFace);
        rulesButton.setTypeface(typeFace);
        startButton.setTypeface(typeFace);
    }
}

File: app/src/main/java/sample/avightclav/checkers/MainActivity.kt
package sample.avightclav.checkers

import android.content.*
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.IBinder
import android.util.Log
import android.widget.Button
import androidx.localbroadcastmanager.content.LocalBroadcastManager

class MainActivity : AppCompatActivity() {

    val TAG = "MainActivity"

    lateinit var saveGameService: SaveGameService
    var bound = false

    val broadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent?) {
            if (intent != null) {
                if (intent.action == SaveGameService.broadcastTag) {
                    val isNotSaving = intent.getIntExtra(SaveGameService.state, -1) == SaveGameService.SAVING_PENDED
                    findViewById<Button>(R.id.resume).isEnabled = !isNotSaving
                }
            }
        }
    }

    val serviceConnection = object : ServiceConnection {
        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
            var binder = service as SaveGameService.SaveGameBinder
            saveGameService = binder.getService()
            bound = true
        }

        override fun onServiceDisconnected(name: ComponentName?) {
            bound = false
        }

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        bindService(Intent(this, SaveGameService::class.java), serviceConnection, Context.BIND_AUTO_CREATE)

        Log.d(TAG, "Started")
        val resumeButton = findViewById<Button>(R.id.resume)

        findViewById<Button>(R.id.start_game_8).setOnClickListener { it ->
            val intent = Intent(this, GameActivity::class.java)
            intent.putExtra("FIELD_SIZE", FieldSize.RUSSIAN)
            startActivity(intent)
        }

        findViewById<Button>(R.id.start_game_10).setOnClickListener { it ->
            val intent = Intent(this, GameActivity::class.java)
            intent.putExtra("FIELD_SIZE", FieldSize.WORLDWIDE)
            startActivity(intent)
        }

        resumeButton.setOnClickListener { it ->
            val intent = Intent(this, GameActivity::class.java)
            intent.putExtra("FIELD_SIZE", FieldSize.RESUME)
            startActivity(intent)
        }

        LocalBroadcastManager.getInstance(this).registerReceiver(broadcastReceiver, IntentFilter(SaveGameService.broadcastTag))
    }

    override fun onStop() {
        super.onStop()

        if (bound) {
            unbindService(serviceConnection)
            bound = false
        }

    }
}
